daemons/ipa-slapi-plugins/ipa-sidgen/ipa_sidgen_common.c

```diff
@@ -554,6 +565,7 @@ int find_sid_for_ldap_entry(struct slapi_entry *entry,
     }
 
 done:
+    slapi_ch_free_string(&uniqueid);
```

Comment:
I think it is likely invalid as 'uniqueid' value is a reference in the slapi_entry. It should not be free, it is the free of the entry that will free it.
Risk of double free.

Reply:
`slapi_entry_attr_get_charptr()` returns malloced memory and explicitly asks for freeing it:
```
char *
slapi_entry_attr_get_charptr(const Slapi_Entry *e, const char *type)
{
    char *p = NULL;
    Slapi_Attr *attr = NULL;

    slapi_entry_attr_find(e, type, &attr);
    if (attr != NULL) {
        Slapi_Value *v;
        const struct berval *bvp;

        slapi_valueset_first_value(&attr->a_present_values, &v);
        bvp = slapi_value_get_berval(v);
        p = slapi_ch_malloc(bvp->bv_len + 1);
        memcpy(p, bvp->bv_val, bvp->bv_len);
        p[bvp->bv_len] = '\0';
    }
    return p;
}
```

Reply:
oppss you are right. Sorry for the noise